packages/mongodb-rag-ingest/src/embed/updateEmbeddedContent.test.ts

```diff
@@ -138,4 +148,139 @@ describe("updateEmbeddedContent", () => {
       "2cbfe9901657ca15260fe7f58c3132ac1ebd0d610896082ca1aaad0335f2e3f1"
     );
   });
+  it("processes chunks concurrently within a page", async () => {
+    const embeddedContentStore = makeMockEmbeddedContentStore();
+    const page: PersistedPage = {
+      ...examplePage,
+      updated: new Date(),
+      action: "updated",
+    };
+
+    // Spy and mock chunkPage
+    const chunkPageSpy = jest
+      .spyOn(chunkPageModule, "chunkPage")
+      .mockResolvedValue([
+        {
+          text: "chunk1",
+          url: "",
+          sourceName: "",
+          tokenCount: 0,
+        },
+        {
+          text: "chunk2",
+          url: "",
+          sourceName: "",
+          tokenCount: 0,
+        },
+        {
+          text: "chunk3",
+          url: "",
+          sourceName: "",
+          tokenCount: 0,
+        },
+      ]);
+
+    const startTimes: number[] = [];
+    const endTimes: number[] = [];
+
+    const mockEmbedder: jest.Mocked<Embedder> = {
+      embed: jest.fn().mockImplementation(async (param) => {
+        const startTime = Date.now();
+        startTimes.push(startTime);
+        await new Promise((resolve) => setTimeout(resolve, 50)); // Simulate delay
+        const endTime = Date.now();
+        endTimes.push(endTime);
+        return { embedding: [1, 2, 3] };
+      }),
+    } as unknown as jest.Mocked<Embedder>;
+
+    await updateEmbeddedContentForPage({
+      embedder: mockEmbedder,
+      store: embeddedContentStore,
+      page,
+      concurrencyOptions: { createChunks: 2 },
+    });
+
+    // Restore chunkPage functionality to avoid affecting other tests
+    chunkPageSpy.mockRestore();
+
+    const embeddedContent = await embeddedContentStore.loadEmbeddedContent({
+      page: examplePage,
+    });
+
+    expect(embeddedContent).toHaveLength(3);
+    const executionPairs = startTimes.map((startTime, i) => ({
+      startTime,
+      endTime: endTimes[i],
+    }));
+
+    // Ensure some overlaps indicating concurrency
+    expect(
+      executionPairs.some((pair, i, pairs) =>
+        pairs.some(
+          (otherPair, j) =>
+            i !== j &&
+            pair.startTime < otherPair.endTime &&
+            otherPair.startTime < pair.endTime
+        )
+      )
+    ).toBe(true);
+  });
+  it("processes pages concurrently", async () => {
+    const pageStore = makeMockPageStore();
+    const concurrentPages: Page[] = [
+      { ...examplePage, url: "https://example.com/test1" },
+      { ...examplePage, url: "https://example.com/test2" },
+      { ...examplePage, url: "https://example.com/test3" },
+    ];
+
+    await persistPages({
+      pages: concurrentPages,
+      store: pageStore,
+      sourceName: "test",
+    });
+
+    const embeddedContentStore = makeMockEmbeddedContentStore();
+
+    const since = new Date("2000-01-01");
+
+    const startTimes: number[] = [];
+    const endTimes: number[] = [];
+
+    const mockEmbedder: jest.Mocked<Embedder> = {
```

Comment:
can you share this mock embedded with the other tests? 

better to not redefine per file